tests/test_wellsmanager.cpp

```diff
@@ -306,3 +306,25 @@ BOOST_AUTO_TEST_CASE(WellSTOPOK) {
     BOOST_CHECK(well_controls_well_is_stopped(ctrls0)); // The first well is closed
     BOOST_CHECK(well_controls_well_is_open(ctrls1));  // The second well is open
 }
+
+BOOST_AUTO_TEST_CASE(removeWellWithNoPerforation) {
+    const std::string filename = "wells_no_perforation.data";
+    Opm::Parser parser;
+    Opm::Deck deck(parser.parseFile(filename));
+    Opm::EclipseState eclipseState(deck);
+    Opm::GridManager gridManager(eclipseState.getInputGrid());
+    const auto& inputGrid = eclipseState.getInputGrid();
+    const Opm::TableManager table ( deck );
+    const Opm::Eclipse3DProperties eclipseProperties ( deck , table, inputGrid);
+    const Opm::Runspec runspec (deck);
+    Opm::Schedule sched(deck, inputGrid, eclipseProperties, runspec);
+
+    const auto eclipseGrid = Opm::UgGridHelpers::createEclipseGrid(*gridManager.c_grid(), inputGrid);
+    sched.filterConnections(eclipseGrid);
```

Comment:
Doesn't this simply remove the connection from the `Schedule` object?  In that case we're not actually testing the new code at all.

Reply:
Yes, but it will not change the status of the well to be `SHUT`. A well is `OPEN`, but have no connection due to the grid processing after parsing, which is exactly what we are testing here. 

If the connections are removed during the parsing period, the well be set to be `SHUT`.

Reply:
Or you can argue that in `filterConnections()`, we should set well to be `SHUT` if there is no connection left.  It is also something we can do.

Reply:
If we do this, then we will throw in `WellsManager_impl.hpp` in stead of giving a warning.